server/webapp/WEB-INF/rails.new/app/controllers/api_v1/users_controller.rb

```diff
@@ -43,12 +37,41 @@ def update
       end
     end
 
+    def create
+      @user_to_operate = user_service.findUserByName(params[:login_name])
+
+      unless @user_to_operate.instance_of?(com.thoughtworks.go.domain.NullUser)
```

Comment:
Checking for existence of user over here is fine, but there could possibly be a race condition. [UserService.update](https://github.com/gocd/gocd/blob/956f8baab3b01eaea52fa345a802e0568933ce0b/server/src/com/thoughtworks/go/server/service/UserService.java#L124) internally calls [saveOrUpdate](https://github.com/gocd/gocd/blob/956f8baab3b01eaea52fa345a802e0568933ce0b/server/src/com/thoughtworks/go/server/service/UserService.java#L367) which does not check if the user already exists. Have a look at [addUserIfDoesNotExist](https://github.com/gocd/gocd/blob/956f8baab3b01eaea52fa345a802e0568933ce0b/server/src/com/thoughtworks/go/server/service/UserService.java#L350) instead.